### PR TITLE
Add AR view controller

### PR DESCRIPTION
**Pull request by xaanimus**

### CODE REVIEW

File: CampusTour/AR/ARExplorerViewController.swift [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    
+    func setItems(items: [ARItemOfInterest]) {
+        itemsOfInterestAndViews = items.map {
+            ItemViewInfo(item: $0,
+                         view: ARItemOfInterestView(item: $0),
+                         node: nil)
+        }
+    }
+    
+    private var sceneView: ARSCNView! = nil
```
Comment (jhseung) [inline CampusTour/AR/ARExplorerViewController.swift]: can we put all the variable declarations at the top?

File: CampusTour/AR/ARExplorerViewController.swift [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        }
+    }
+    
+    func startAr() {
+        let trackingConfig = ARWorldTrackingConfiguration()
+        trackingConfig.worldAlignment = .gravityAndHeading
+        sceneView.session.run(trackingConfig, options: [.resetTracking, .removeExistingAnchors])
+    }
+    
+    func stopAr() {
```
Comment (jhseung) [inline CampusTour/AR/ARExplorerViewController.swift]: Will there be sth here?

File: CampusTour/AR/VirtualView.swift [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        
+        self.addSubview(wrapperView)
+        wrapperView.snp.makeConstraints {
+            $0.center.equalToSuperview()
+            $0.width.lessThanOrEqualToSuperview()
+            $0.height.lessThanOrEqualToSuperview()
+        }
+        
+        self.setNeedsLayout()
+        self.layoutIfNeeded()
```
Comment (jhseung) [inline CampusTour/AR/VirtualView.swift]: isn't only layoutifneeded necessary?

File: CampusTour/AR/VirtualView.swift [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    let location: CLLocation
+}
+
+class ARItemOfInterestView : UIView {
+    var subtitleLabel: UILabel?
+    
+    func updateSubtitleWithDistance(meters: Double) {
+        subtitleLabel?.text = String.init(format: "%.1f m", meters)
+        self.setNeedsLayout()
+        self.layoutIfNeeded()
```
Comment (jhseung) [inline CampusTour/AR/VirtualView.swift]: same here

**Comment by jhseung** [comment]
Looks good to go--note my comments 😃

File: CampusTour/AR/ARExplorerViewController.swift (hunk 2 above)
Comment (xaanimus) [reply CampusTour/AR/ARExplorerViewController.swift]: there should be some cleanup, I haven't gotten to that yet

File: CampusTour/AR/VirtualView.swift (hunk 3 above)
Comment (xaanimus) [reply CampusTour/AR/VirtualView.swift]: is needsLayout isn't set then layoutIfNeeded won't do anything

**Pull request merged**